src/lib.rs

```diff
@@ -22,7 +22,7 @@
 //! {
 //!     // get first `name` element
 //!     let name = names_element.get_mut_child("name").expect("Can't find name element");
-//!     name.attributes.insert("suffix".to_owned(), "mr".to_owned());
+//!     name.attributes.insert(AttributeName::local("suffix"), "mr".to_owned());
```

Comment:
I wonder if we can't somehow wrap `AttributeMap` (i.e. make it a real struct instead of an alias, even if it's just a newtype struct) which would make this change less painful for downstreams. If we did that, we could (I think, running on fumes here):

```rust
trait IntoAttributeKey {
    fn into_attribute_key(self) -> AttributeName;
}

pub struct AttributeMap<K: IntoAttributeKey, V>(AttributeMapInner<K, V>);

impl<S: AsRef<str>> IntoAttributeKey for S {…}
```

That was my main concern with taking a similar approach in #33.